main.py

```diff
@@ -465,38 +468,47 @@ def performEvaluation(issue_data) -> Result:
 
         jdk_name = f"amazon-corretto-{version}"
         jdk_tar_name = f"{jdk_name}.tar.gz"
-        jdk_abs_path = os.path.abspath(jdk_tar_name)
-        if not os.path.exists(jdk_abs_path):
-            download_with_wget(jdk_url, jdk_tar_name)
-        
-        extracted_jdk_name = jdk_name + "."+ "jdk"
-        extracted_jdk_abs_path = os.path.abspath(extracted_jdk_name)
-        if os.path.exists(extracted_jdk_abs_path):
-            shutil.rmtree(extracted_jdk_abs_path)
-        if os.path.exists(jdk_abs_path):
-            extract_tar_gz(jdk_abs_path)
+        jdk_tar_abs_path = os.path.abspath(jdk_tar_name)
+
+        if not os.path.exists(jdk_tar_abs_path):
+            download_with_wget(jdk_url, jdk_tar_abs_path)
         
-        set_directory_exec_permission(extracted_jdk_abs_path)
+        extracted_jdk_name = ""
+        if op == "linux":
+            extracted_jdk_abs_path = os.path.abspath(jdk_name) #/../amazon-corretto-8
+        else:
+            extracted_jdk_abs_path = os.path.abspath(jdk_name) + ".jdk" #//.//amazon-corretto-8.jdk
 
-        jdk_home = os.path.join(extracted_jdk_abs_path, "Contents", "Home")
-        os.environ["JAVA_HOME"] = jdk_home 
+        if not os.path.exists(extracted_jdk_abs_path):
+            os.makedirs(extracted_jdk_abs_path, exist_ok=True)
+            if os.path.exists(extracted_jdk_abs_path):
+                extract_and_rename(jdk_tar_abs_path, extracted_jdk_abs_path)
+        
+        if op == "linux":
+            java_path = os.path.join(extracted_jdk_abs_path, "bin", "java")
+        else:
```

Comment:
This should be an else if, so that running on a non-standard OS doesn't silently try to use the mac config (raise an exception instead)

Reply:
```
op_sys = platform.system()
    if op_sys != "Linux" and op_sys != "Darwin":
        print(f"{op_sys} no supported")
        sys.exit(1)
```
This check added at main method. Please allow me to add this exception later.

Reply:
Even so, it's risky to use `else:` when you really mean `elif on macos:` - that other line of code might be removed in the future.